sktime/decomposition/vampnet.py

```diff
@@ -0,0 +1,662 @@
+from ..util import module_available
+
+if not module_available("torch"):
+    raise RuntimeError("Tried importing VampNets; this only works with a PyTorch installation!")
+del module_available
+
+from typing import Optional, Union, List, Callable, Tuple
+
+import numpy as np
+
+from ..base import Transformer, Model, Estimator
+from ..data.util import TimeSeriesDataset
+
+import torch
+import torch.nn as nn
+from torch.utils.data import Dataset, DataLoader
+
+
+def sym_inverse(mat, epsilon: float = 1e-6, ret_sqrt=False, mode='regularize'):
+    """ Utility function that returns the inverse of a matrix, with the
+    option to return the square root of the inverse matrix.
+
+    Parameters
+    ----------
+    mat: numpy array with shape [m,m]
+        Matrix to be inverted.
+    epsilon : float
+        Cutoff for eigenvalues.
+    ret_sqrt: bool, optional, default = False
```

Comment:
you can keep this as is but my first read of this kw was "retain square root" not "return" and i would've imagined w/out reading the docs that i'm choosing between 1 and 2 things returned

Reply:
fair, suggestions for a better name?

Reply:
`return_sqrt` ?

Reply:
fixed